external/dhcpd/dhcpd_lwip.c

```diff
@@ -223,9 +226,24 @@ int dhcp_server_start(char *intf, dhcp_sta_joined dhcp_join_cb)
 	memcpy(data->intf, intf, strlen(intf) + 1);
 	data->fn = dhcp_join_cb;
 
-	int ret = ERROR;
 	g_dhcpd_term = 0;
-	ret = pthread_create(&g_dhcpd_tid, NULL, _dhcpd_join_handler, (void *)data);
+		/* Initialize the attribute variable */
+
+	if ((ret = pthread_attr_init(&attr)) != 0) {
+		ndbg("Failed to init attr\n");
+		free(data->intf);
```

Comment:
How about using 'goto' to line 234, 242, and 249 to make code simpler?
ex)
int dhcp_server_start(char *intf, dhcp_sta_joined dhcp_join_cb)
{	
    int ret = OK;
    ...
    if ((ret = ~~) != OK) {
        ...
        ret = ERROR;
        goto error_out:    // Or free_out:?
    }
    ...
error_out:
    free(data->intf);
    free(data);
    return ret;
}